src/internal/auth/interceptor.go

```diff
@@ -250,25 +281,48 @@ func (i *Interceptor) InterceptUnary(ctx context.Context, req interface{}, info
 		return nil, fmt.Errorf("no auth function for %q, this is a bug", info.FullMethod)
 	}
 
-	if err := a(pachClient, info.FullMethod); err != nil {
-		logrus.Errorf("denied unary call %q\n", info.FullMethod)
+	username, err := a(pachClient, info.FullMethod)
+
+	if err != nil {
+		logUser := "unauthenticated"
+		if username != "" {
+			logUser = username
+		}
+		logrus.Errorf("denied unary call %q to user %v\n", info.FullMethod, logUser)
 		return nil, err
 	}
+
+	if username != "" {
+		ctx = context.WithValue(ctx, WhoAmIResultKey, username)
+	}
+
 	return handler(ctx, req)
 }
 
 // InterceptStream applies authentication rules to streaming RPCs
 func (i *Interceptor) InterceptStream(srv interface{}, stream grpc.ServerStream, info *grpc.StreamServerInfo, handler grpc.StreamHandler) error {
-	pachClient := i.env.GetPachClient(stream.Context())
+	ctx := stream.Context()
+	pachClient := i.env.GetPachClient(ctx)
 	a, ok := authHandlers[info.FullMethod]
 	if !ok {
 		logrus.Errorf("no auth function for %q\n", info.FullMethod)
 		return fmt.Errorf("no auth function for %q, this is a bug", info.FullMethod)
 	}
 
-	if err := a(pachClient, info.FullMethod); err != nil {
-		logrus.Errorf("denied streaming call %q\n", info.FullMethod)
+	username, err := a(pachClient, info.FullMethod)
+
+	if err != nil {
+		logUser := "unauthenticated"
+		if username != "" {
+			logUser = username
+		}
```

Comment:
it might be possible to dry this up a little with a function, not really a big deal

```
func nameOrUnauthenticated(name string) string {
    if name == "" {
        return unauthenticated
    }
    return name
}
```